packages/input_schema/src/schema.json

```diff
@@ -32,22 +32,80 @@
             "type": "object",
             "patternProperties": {
                 "^": {
-                    "oneOf": [
-                        { "$ref": "#/definitions/stringProperty" },
-                        { "$ref": "#/definitions/stringEnumProperty" },
-                        { "$ref": "#/definitions/arrayProperty" },
-                        { "$ref": "#/definitions/objectProperty" },
-                        { "$ref": "#/definitions/integerProperty" },
-                        { "$ref": "#/definitions/booleanProperty" },
-                        { "$ref": "#/definitions/anyProperty" }
-                    ]
+                    "if": {
+                        "properties": {
+                            "resourceType": {
+                                "not": {
+                                    "type": "string"
+                                }
+                            }
+                        }
+                    },
+                    "then": {
+                        "oneOf": [
+                            { "$ref": "#/definitions/stringProperty" },
+                            { "$ref": "#/definitions/stringEnumProperty" },
+                            { "$ref": "#/definitions/arrayProperty" },
+                            { "$ref": "#/definitions/objectProperty" },
+                            { "$ref": "#/definitions/integerProperty" },
+                            { "$ref": "#/definitions/booleanProperty" },
+                            { "$ref": "#/definitions/anyProperty" }
+                        ]
+                    },
+                    "else": {
+                        "oneOf": [
+                            { "$ref": "#/definitions/resourceProperty" },
+                            { "$ref": "#/definitions/resourceArrayProperty" }
+                        ]
+                    }
                 }
             }
         }
     },
     "additionalProperties": false,
     "required": ["title", "type", "properties", "schemaVersion"],
     "definitions": {
+        "resourceProperty": {
+            "title": "Resource property",
```

Comment:
Nit: It looks the title, description, nullable, sectionCaption, sectionDescription appear in all definition. Could we move these to special definitions? 
```
{
...
    "definitions": {
        "baseProperty": {
            "type": "object",
            "properties": {
                "title": {"type": "string"},
                "description": {
                    "type": "string"
                },
                "nullable": { "type": "boolean"}
            },
            "required": ["title", "description"]
        },
        "resourceProperty": {
            "allOf": [
                {"$ref": "#/definitions/baseProperty"},
                {
                    "properties": {
                        ...the rest of props on top of baseProperty
                    }
                }
            ],
            "required": ["type", "title", "description", "resourceType"]
        },
        ...
    }
...
}
```